.travis.yml

```diff
@@ -56,12 +62,23 @@ addons:
     - libatlas-base-dev
 
 before_install:
+  - SECONDS=0 && function timeout() { perl -e 'alarm shift; exec @ARGV' "$@"; }
```

Comment:
The entire point of this is to make sure that the cache is always uploaded, even if the tests time out. This ensures robust and efficient operation of the tests because each time we need to set up another osx instance is up to 10 minutes of installation time if the packages are not cached and 3 minutes if they are.